map.go

```diff
@@ -14,17 +14,33 @@ type Map[K, V any] struct {
 	prefix []byte
 	sk     sdk.StoreKey
 
+	keyName  string
+	name     string
 	typeName string
 }
 
-func NewMap[K, V any](sk sdk.StoreKey, namespace Namespace, kc KeyEncoder[K], vc ValueEncoder[V]) Map[K, V] {
+func NewMap[K, V any](schema Schema, namespace Prefix,
+	keyName string, kc KeyEncoder[K],
+	valueName string, vc ValueEncoder[V]) Map[K, V] {
```

Comment:
wondering what valueName represents and why ValueEncoder.Type is not enough as a value identifier

Reply:
It's something to think through. Basically think about how you would refer to these things in a genesis file or query. Say it were a map from int -> int. You wouldn't want to call the collection "ints". Maybe it's a mapping from account ID to account sequence or something, so you would say key name is account_id and value name is account_seq, and maybe the whole collection would then be called account_seq. Not sure what is best really but hopefully that's helpful context

Reply:
An alternative would just be naming the collection (ex. `account_se_map`and then calling things simply `key` and `value` in json, ex:

```
// genesis.json
{
  "account_seq_map":[
     {"key":10, "value":100},
     {"key":15, "value":9},
     ...
  ]
}
```

Reply:
yes I get what you mean, it's something I was thinking when porting the core in the sdk, I associated it with prefix and called it "humanized" prefix: https://github.com/cosmos/cosmos-sdk/pull/14134/files#diff-3fd3431195e40cb774e422147a847310792d01eba9d8263f512d027fc8ea059dR48

Reply:
do you think one single "human" name defining the single collection is enough, or do you think we need to define human names for both key and value pairs.

Reply:
> do you think one single "human" name defining the single collection is enough, or do you think we need to define human names for both key and value pairs.

I'm not sure. It is nice in the orm that you know things are called `address`, `amount`, etc. from the field names and then that can propagate all the way to queries (https://github.com/cosmos/cosmos-sdk/pull/13438). But maybe `key` and `value` are okay for maps, and the orm behavior can be gotten with a `collections.Table` abstraction